### PR TITLE
enable `typed: strict` in Homebrew::CLI::Args

### PR DESCRIPTION
**Pull request by dduugg**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Part of https://github.com/Homebrew/brew/issues/17297

Also helpful as I hack at removing the `OpenStruct` dependency

### CODE REVIEW

File: Library/Homebrew/cli/args.rb [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 # frozen_string_literal: true
 
 require "ostruct"
 
 module Homebrew
   module CLI
     class Args < OpenStruct
+      # FIXME: Enable cop again when https://github.com/sorbet/sorbet/issues/3532 is fixed.
+      # rubocop:disable Style/MutableConstant
+      OptionsType = T.type_alias { T::Array[[String, T.nilable(String), T.nilable(String), String, T::Boolean]] }
```
Comment (carlocab) [inline Library/Homebrew/cli/args.rb:11]: Could use a comment describing what this is, since I don't know how to parse the block inside `T.type_alias`.

**Review by carlocab — APPROVED** [review]

File: Library/Homebrew/cli/args.rb (hunk 1 above)
Comment (dduugg) [reply Library/Homebrew/cli/args.rb:11]: Good suggestion! Here's what I have so far:
```ruby
      # Represents a processed option. The elements are:
      # 0: short option name (e.g. "-d")
      # 1: long option name (e.g. "--debug")
      # 2: 
      # 3: option description (e.g. "Print debugging information")
      # 4: whether the option is hidden)```
```
I can't figure out the element at index 2, it was `nil` for every combination of command and options i threw at it.

Comment (carlocab) [reply Library/Homebrew/cli/args.rb:11]: Interesting. Does Sorbet throw a (static or runtime) error if we replace `2` with `NilClass`?

Also, I wonder if we could/should use a `T::Struct` here instead.

**Review by MikeMcQuaid — APPROVED** [review]
Looks good thanks @dduugg! Will hold off merging until after next tag 🔜 today.

File: Library/Homebrew/cli/args.rb (hunk 1 above)
Comment (Bo98) [reply Library/Homebrew/cli/args.rb]: Looks like it's never used anyway so we could remove it:

https://github.com/Homebrew/brew/blob/2a5eb025f1d36e41667bb22a564ce03cb1eceb48/Library/Homebrew/commands.rb#L202

https://github.com/Homebrew/brew/blob/2a5eb025f1d36e41667bb22a564ce03cb1eceb48/Library/Homebrew/manpages.rb#L101

Comment (dduugg) [reply Library/Homebrew/cli/args.rb]: It's been yanked, PTAL

**Review by Bo98 — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

File: Library/Homebrew/cli/args.rb [hunk 2]
```diff
@@ -165,10 +186,12 @@ def cli_args
         @cli_args.freeze
       end
 
-      def respond_to_missing?(method_name, *)
+      sig { params(method_name: Symbol, _include_private: T::Boolean).returns(T::Boolean) }
+      def respond_to_missing?(method_name, _include_private = false)
```
Comment (carlocab) [inline Library/Homebrew/cli/args.rb:190]: Not sure why it didn't error here, but it's now erroring on `master`:

https://github.com/Homebrew/brew/actions/runs/10469292764/job/28991992536#step:7:14

Comment (carlocab) [reply Library/Homebrew/cli/args.rb:190]: #18097

**Pull request merged**